mockgcp/mockapikeys/apikeysv2.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockapikeys
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	longrunning "google.golang.org/genproto/googleapis/longrunning"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/api/apikeys/v2"
+)
+
+type APIKeysV2 struct {
+	*MockService
+	pb.UnimplementedApiKeysServer
+}
+
+func (s *APIKeysV2) GetKey(ctx context.Context, req *pb.GetKeyRequest) (*pb.Key, error) {
+	name, err := s.parseAPIKeyName(req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.Key{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "key %q not found", name)
+		} else {
+			return nil, status.Errorf(codes.Internal, "error reading key: %v", err)
+		}
+	}
+
+	return obj, nil
+}
+
+func (s *APIKeysV2) GetKeyString(ctx context.Context, req *pb.GetKeyStringRequest) (*pb.GetKeyStringResponse, error) {
```

Comment:
Just for self-education purpose: How do we know that this additional function is needed? Is it only detectable by test failures?

Reply:
Right - we implement the methods we need to pass the tests.  When we're missing one, we see an error like `"method GetKeyString not implemented"` in the logs, coming from pb.UnimplementedApiKeysServer